### PR TITLE
[PT FE]: supports eye, tril, triu, tensor.fill_ and implicit scalars

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *added support aten::eye, aten::tril, aten::triu, aten.fill_, aten::ScalarImplicit, aten::IntImplicit*
 - *...*

### Tickets:
 - *101091*
 - *101073*
 - *101098*
 - *99222*

### CODE REVIEW

File: src/frontends/pytorch/src/op/trilu.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+namespace base {
+
+OutputVector translate_base_triu_tril(NodeContext& context, bool upper) {
+    auto input_tensor = context.get_input(0);
+    auto input_shape = context.mark_node(std::make_shared<opset10::ShapeOf>(input_tensor));
+    auto zero = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {0}));
+    auto one = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {1}));
+    auto minus_one = context.mark_node(opset10::Constant::create(element::i32, Shape{}, {-1}));
+    auto minus_two = context.mark_node(opset10::Constant::create(element::i32, Shape{}, {-2}));
+    const auto m = context.mark_node(std::make_shared<opset10::Gather>(input_shape, minus_one, zero));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/trilu.cpp:22]: Is it necessary to mix `i32` and `i64` types here?

File: src/frontends/pytorch/src/op/eye.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset10.hpp"
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/op/eye.cpp]: updated

File: src/frontends/pytorch/src/op/trilu.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/trilu.cpp]: aligned

**Review by mvafin — APPROVED** [review]

File: src/frontends/pytorch/src/op/eye.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    } else if (num_inputs == 6) {
+        // aten::eye(n, m, dtype, layout, device, pin_memory)
+        y = context.get_input(1);
+        y = context.mark_node(std::make_shared<ov::op::v0::Convert>(y, element::i64));
+        dtype_id = 2;
+    } else {
+        FRONT_END_OP_CONVERSION_CHECK(false, "Unsupported number of inputs: ", num_inputs, " for aten::eye");
+    }
+    if (!context.input_is_none(dtype_id)) {
+        dtype = convert_dtype(dtype_id);
```
Comment (eaidova) [inline src/frontends/pytorch/src/op/eye.cpp:40]:
```suggestion
        dtype = convert_dtype(context.const_input<int64_t>(dtype_id));
```

File: src/frontends/pytorch/src/op/eye.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        y = context.get_input(1);
+        y = context.mark_node(std::make_shared<ov::op::v0::Convert>(y, element::i64));
+        dtype_id = 2;
+    } else {
+        FRONT_END_OP_CONVERSION_CHECK(false, "Unsupported number of inputs: ", num_inputs, " for aten::eye");
+    }
+    if (!context.input_is_none(dtype_id)) {
+        dtype = convert_dtype(context.const_input<int64_t>(dtype_id));
+    }
+    auto eye = context.mark_node(std::make_shared<ov::op::v9::Eye>(x, y, diagonal, element::i64));
```
Comment (eaidova) [inline src/frontends/pytorch/src/op/eye.cpp:42]:
```suggestion
    auto eye = context.mark_node(std::make_shared<ov::op::v9::Eye>(x, y, diagonal, element::i32));
```

**Pull request merged**